crates/forge_domain/src/chat_response.rs

```diff
@@ -12,4 +12,13 @@ pub enum ChatResponse {
     ToolCallEnd(ToolResult),
     Usage(Usage),
     Event(Event),
+    /// Represents a retry attempt due to a retriable error
+    Retry {
+        /// The error that caused the retry
+        error: RetryError,
```

Comment:
```suggestion
        reason: String,
```

use `reason` to specify "why" are we retrying.